amulet/sentry.py

```diff
@@ -307,28 +344,152 @@ def wait(self, timeout=300):
 
         Raises an error if the timeout is exceeded.
         """
-        ready = False
-        try:
-            with helpers.timeout(timeout):
-                while not ready:
-                    for unit in self.unit.keys():
-                        status = self.unit[unit].juju_agent()
-
-                        # Check if we have a hook key and it's not None
-                        if status is None:
-                            ready = False
-                            break
-                        if 'hook' in status and status['hook']:
-                            ready = False
-                            break
-                        else:
-                            ready = True
-        except:
-            raise
+        def check_status():
+            status = self.get_status()
+            for service_name in self.service_names:
+                service = status.get(service_name, {})
+                for unit_name, unit in service.items():
+                    if unit['workload-status']:
+                        if unit['workload-status']['current'] not in ('active', 'unknown'):
```

Comment:
This is wrong.  We should only be blocking on agent-status to match the previous behavior of wait.  Waiting on workload-status should be handled entirely by wait_for_messages.

Specifically, this breaks testing partial deployments that set blocked status, which would otherwise work in the old version.